src/Raven.Server/Documents/TombstoneCleaner.cs

```diff
@@ -118,9 +120,50 @@ internal async Task<long> ExecuteCleanup(long? numberOfTombstonesToDeleteInBatch
             return numberOfTombstonesDeleted;
         }
 
+        private void CreateWarningIfThereAreBlockingTombstones(HashSet<string> tombstoneCollections)
+        {
+            var currentBlockingTombstones = new Dictionary<(string, string), long>();
+            var tombstonesPerCollection = new Dictionary<string, long>(StringComparer.OrdinalIgnoreCase);
+            bool needToWarn = false;
+
+            foreach (var subscription in _subscriptions)
+            {
+                var disabledSubscribers = subscription.GetDisabledSubscribersCollections(tombstoneCollections);
+                foreach (var disabledSubscriber in disabledSubscribers.Keys)
+                {
+                    HashSet<string> blockedTombstoneCollections = disabledSubscribers[disabledSubscriber];
+
+                    using (_documentDatabase.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                    using (context.OpenReadTransaction())
```

Comment:
Let's open a read transaction only you need to fetch the number of tombstones

Reply:
done.